base/logging.jl

```diff
@@ -502,7 +513,7 @@ with_logger(logger) do
 end
 ```
 """
-with_logger(f::Function, logger::AbstractLogger) = with_logstate(f, LogState(logger))
+with_logger(@nospecialize(f::Function), logger::AbstractLogger) = with_logstate(f, LogState(logger))
```

Comment:
I would not expect this to make a difference; it doesn't affect inference and we will already avoid specializing this function on `f`.

Reply:
Yeah, it doesn't seem to change anything.